3_javascript/javascript1/week2/script.js

```diff
@@ -0,0 +1,134 @@
+// getFullName function
+const getFullname = (
+  firstname,
+  surname,
+  useFormalName = false,
+  areYouMale = true
+) => {
+  if (
+    (firstname !== "" && surname !== "" && firstname.trim() !== "",
+    surname.trim() !== "")
+  ) {
+    if (useFormalName && areYouMale) {
+      console.log("lord " + firstname + " " + surname);
+    } else if (useFormalName && !areYouMale) {
+      console.log("lady " + firstname + " " + surname);
+    } else {
+      console.log(firstname + " " + surname);
+    }
+  } else {
+    console.log("provide a valid value for firstname and surname");
+  }
+};
+getFullname("Niloufar", "Khaefi", true, false);
+getFullname("Mohammad", "Basiri", true);
+getFullname("Armin", "Basiri", false);
+getFullname(" ", " ", true);
+
+// Event Application
+const getEventDay = (daysUntilEvent) => {
+  const weekdays = [
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+  const today = new Date();
+  const dayIndex = today.getDay();
+  const eventDayIndex = (dayIndex + daysUntilEvent) % 7;
+  console.log(`The event will be held on a ${weekdays[eventDayIndex]}`);
+};
+getEventDay(9);
```

Comment:
Nice and clean solution, the only thing is that the function should return the value, not print it out :) 

```
// With todays weekday a tuesday
console.log(getEventWeekday(9)); // Logs out "Thursday"
```